client_error.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Generic error thrown by the SDK.
+ * @extends {Error}
+ */
+class ClientError extends Error {
+  /**
+   * @override
+   * @param {StatusCode} code error status code
+   * @param {string} args
+   */
+  constructor(code, ...args) {
+    super(...args);
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, ClientError);
+    }
+    this.name = 'ClientError';
+
+    /**
+     * @type {StatusCode}
+     */
+    this.code = code;
+  }
+
+  /**
+   * @return {StatusCode}
+   */
+  getStatusCode() {
```

Comment:
Sorry, I realized that having a getter for just status code is a bit weird for consistency, so let's remove this function and make users use err.code like err.message.